File: AOJ/0328/0328.c
#include<stdio.h>

#define NMAX 100000
#define PMAX 100000

int N, T[2*PMAX+20+1];

int main(){
  int i, a, b;
  scanf("%d", &N);
  for ( i = 0; i < N; i++ ){
    scanf("%d %d", &a, &b);
    T[a+b]++;
  }
  for ( i = 0; i < 2*PMAX+20; i++ ){
    T[i+1] += T[i]/2;
    T[i] = T[i]%2;
    if ( T[i] ) printf("%d %d\n", i, 0);
  }

  return 0;
}

File: AOJ/0328/0328.py
#include<iostream>
using namespace std;
static const int NMAX = 100000;
static const int PMAX = 100000;

int N, T[2*PMAX+20+1];

int main(){
  int a, b;
  cin >> N;
  for ( int i = 0; i < N; i++ ){
    cin >> a >> b;
    T[a+b]++;
  }
  for ( int i = 0; i < 2*PMAX+20; i++ ){
    T[i+1] += T[i]/2;
    T[i] = T[i]%2;
    if ( T[i] ) cout << i << " " << 0 << endl;
  }

  return 0;
}
